### PR TITLE
feat(ui): Added theming support

### PR DESCRIPTION
**Pull request by lupusA**
Hi, 
this PR adds a preliminary theming support. Themes are defined in CSS and can be selected at the top right of the UI. 

Note that this PR is currently a draft. The selection is not persistent and there is still some work to do. This PR also contains some minor code refactorings. 

https://user-images.githubusercontent.com/37236531/236008253-51d64f2d-3da2-4fcf-b103-fb8c6e0af31c.mp4

This PR closes #118

Cheers,

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,10 +1,9 @@
 import axios from 'axios';
 import 'bootstrap-dark-5/dist/css/bootstrap-nightshade.min.css';
```
Comment (jkowalski) [inline src/App.jsx:2]: do we still need that?

File: src/Theme.css [hunk 2]
```diff
@@ -0,0 +1,68 @@
+/*Supported themes */
+.light-theme {
```
Comment (jkowalski) [inline src/Theme.css:2]: i'm not an expert in CSS, what's the browser support for variables? is it universal at this point?

File: src/App.jsx (hunk 1 above)
Comment (lupusA) [reply src/App.jsx:2]: It is still used for the CSRF token - line 35. Bootstrap itself is used within the wohle UI.

File: src/Theme.css (hunk 2 above)
Comment (lupusA) [reply src/Theme.css:2]: CSS variables are widely supported. The only browser that doesn't support variables is Microsofts IE. However, support for the Internet Explorer is running out anyways. Do we need to support it?

File: src/App.jsx (hunk 1 above)
Comment (lupusA) [reply src/App.jsx]: bootstrap-dark has been removed

**Comment by jkowalski** [comment]
I've tried this and it is very nice feature, but we need some tweaks:

1. When I open the UI for the first time I get black-on-black experience:

image

My old `ui-preferences.json` had:

```{"bytesStringBase2":false,"theme":"dark","pageSize":50}```

Switching the theme to something else and switching back fixed the issue but it is annoying. Can we remap `dark` to `dark-theme` and `light` to `light-theme` when we fetch the preferences for backwards compatibility?

2. The font size for the drop-down is a tiny bit too large compared to other fonts - can you make it a notch smaller?

image

3. Also, would it make sense to introduce `Preferences` tab and move theme selector over there? I'm not sure the dropdown belongs on every single page.

4. Pastel scheme has black-on-dark red `danger` button which is hard to read, should it be white-on-red or use lighter red?

image

5. Section headers in `pastel` theme are too subtle pink on white which looks odd, perhaps make it darker purple:

image

7. The expanded dropdown weirdly overlaps with `Repository/Update Description` button:

image

**Comment by lupusA** [comment]
Hey @jkowalski, 

will work on it on the weekend. 

Cheers,

**Comment by lupusA** [comment]
Hey @jkowalski, 

will work on it on the weekend. 

Cheers,

**Comment by lupusA** [comment]
Hi @jkowalski, 

i worked in the preferences tab. Would you mind to test it?

Cheers,

**Comment by jkowalski** [comment]
This is looking very promising, i found couple issues:

1. When i hover over a link in dark mode, the text is dark-on-dark and unreadable:

image

2. Can we make the dark mode background closer to what it was before, right now the gray is quite bright.

3. In the preferences panel: 

"Bytes to the base of 2"

Can we convert this to a 2-item drop-down or a radio button:

```
Size Display:
[x] Use base-10 suffixes (KB, MB, GB, TB)
[ ] Use base-2 suffixes (KiB, MiB, GiB, TiB)
```

4. Now that we have `Page Size` in preferences, can we make it editable and remove all the other page selectors froma actual tables?

5. As for the theme selection, can we make the dropdown narrower? Perhaps make the whole preferences panel 1/2 or  1/3 width until we have more controls to add there?

**Comment by lupusA** [comment]
Hey @jkowalski, 

thanks for the feedback. The points you listed can easily be fixed. I will fix them in the next commit. Maybe, we need to rebase after that 😄 

Considering point 4, I would argue that we keep it as a readonly preferences. I would find it cumbersome to switch back to the preferences tab to just change the pagination settings. 

It is more convenient to set the page size below the actual table, when there are too many items. 

What do you think?

Cheers,

**Comment by lupusA** [comment]
Hi @jkowalski, 

point 5 is still open. The other points have been incorporated in the latest commit and can be tested. 

Cheers,

**Comment by lupusA** [comment]
Hi @jkowalski, 

i will finalize this pr on the weekend. There are some minor things to be solved related to the dark theme. 

Would you mind checking if this PR than can be merged?

Cheers

**Comment by jkowalski** [comment]
This is great, we could probably improve theme selector a bit (icons instead of names), but i'm going to merge this and integrate into Kopia so folks can try it out.

**Pull request merged**